src/org/sosy_lab/common/NativeLibraries.java

```diff
@@ -105,55 +111,62 @@ public static Architecture guessVmArchitecture() {
         return currentArch;
       }
 
-      @Var String prop = System.getProperty("os.arch.data.model");
-      if (isNullOrEmpty(prop)) {
-        prop = System.getProperty("sun.arch.data.model");
+      currentArch = downgradeArchIfRequired(getArchFromSystemProperty());
+
+      return currentArch;
+    }
+
+    private static Architecture getArchFromSystemProperty() {
+      @Var String osArch = StandardSystemProperty.OS_ARCH.value();
+      if (isNullOrEmpty(osArch)) {
+        throw new UnsatisfiedLinkError("No value for os.arch" + REPORT_MESSAGE);
       }
 
-      if (!isNullOrEmpty(prop)) {
-        switch (prop) {
-          case "32":
-            currentArch = Architecture.X86;
-            break;
-          case "64":
-            currentArch = Architecture.X86_64;
-            break;
-          default:
-            throw new UnsatisfiedLinkError(
-                "Unknown value for os.arch.data.model: '"
-                    + prop
-                    + "', please report this together with information about your system "
-                    + "(OS, architecture, JVM).");
-        }
-      } else {
+      osArch = Ascii.toLowerCase(osArch.replace(" ", ""));
+
+      switch (osArch) {
+        case "i386":
+        case "i686":
+        case "x86":
+          return Architecture.X86;
+        case "amd64":
+        case "x86_64":
+          return Architecture.X86_64;
+        case "arm":
+        case "armv6":
+        case "armv7":
```

Comment:
What are the differences between these? Is it safe to ignore them? Could this cause problems in the future? If we just return `ARM` now, we cannot change the later if we need to distinguish between them.

The questions to answer would be "Can `arm` binary libraries be loaded into a `armv7` JVM?" etc.

For `i386`, `i686`, and `x86` it is clear that these are almost synonyms and that the differences (which affect only processors from the nineties, so can be ignore. But for ARM I don't know.

Do you have a reference documentation for these values? Would be good to link it.

Reply:
Each ARM generation has its own architecture. Of course, there are differences between ARM 6, 7, 8, 9. 
There is one major architecture "AArch64" (ARM64) that is available on Armv8-A onwards, which was extended for ARM 9. I will remove older ARM versions from the code, as even ARM 7 is 20 years old and might no longer be available in devices. I do not have any matching hardware (except ARM8), so I can not provide other examples or testing.

When looking into GCC documentation, there axists a broad range of options and values (e.g., see https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html under option `-march`).
The binaries of Z3 are compiled with a plain option `arch=arm64` (https://github.com/Z3Prover/z3/blob/a3f35b6830fb9639a3e1d0dc23250e91973cb244/scripts/nightly.yaml#L124) and do not seem to include further architecture specifications.